### DISCUSSION TITLE
Webhooks are not fired

### DISCUSSION THREAD

**Discussion by ricccrd** [juspay/hyperswitch#3967] [Ideas / Feature Requests]
What up fellas!

Great work on hyperswitch. Love it.

Guys could you help me out, Im trying to get webhooks working.
Im in the test mode, but seems that webhook requests are not coming in.

Are there any prerequisites? do webhooks work in test mode?
What should I check.

With this big feature request. 
Make UI for requests fired, just like stripe/github has. That is great for debugging.

Any help appreciated! Thanks.
PS, sandbox, cloud-hosted

**Comment by SanchithHegde** [comment]
Hey @huttarichard!

> Guys could you help me out, Im trying to get webhooks working. Im in the test mode, but seems that webhook requests are not coming in.
> 
> Are there any prerequisites? do webhooks work in test mode? What should I check.

Yes, webhooks do work in our sandbox environment. For information on setting up Hyperswitch to send webhooks to your application, please refer to the docs (if you haven't already):
https://docs.hyperswitch.io/hyperswitch-cloud/webhooks

Note that there's a minor change in the location in the control center where you can configure the webhook URL: you can find it under Developers > Payment Settings.

Also, could you try configuring a `smee.io` URL for the webhook URL in the control center, and check if you receive any webhooks there?

> With this big feature request.
> Make UI for requests fired, just like stripe/github has. That is great for debugging.

I agree that would be useful for debugging, could you please raise a feature request for this on our Discussions space, so that we can track it independently?

**Comment by ricccrd** [comment]
@SanchithHegde Thank you for answer!

Yeah, that is wierdest thing ever.

Try:
https://www.toptal.com/developers/postbin/

While https://smee.io/ and https://pipedream.com/ works, toptal doesnt nor my endpoint.

I will try to investigate further.
But if you could give it a try, that would be highly appreciated.

**Comment by SanchithHegde** [comment]
@huttarichard Our sandbox environment has a set of "allowed" URLs that the application is permitted to interact with. `smee.io` and `pipedream.com` were allowed previously, while `toptal.com/developers/postbin` was not. We've now allowed `toptal.com/developers/postbin` as well, so that should work, but your application endpoint would obviously not.

Will be closing this discussion as resolved.

**Reply by ricccrd** [reply]
I see, well in that regards, toptal is not important to me. What is new to me, is fact you guys have to whitelist domains.
That is quite confusing because you do not find this in docs, at least I did not.

However for the implementation it is somewhat necessary. I'll run it locally.
Thanks.

**Reply by SanchithHegde** [reply]
> What is new to me, is fact you guys have to whitelist domains.
> That is quite confusing because you do not find this in docs, at least I did not.

The domain whitelisting is specific to our hosted sandbox environment and has been set up with the infrastructure, that is not something the application is aware of.

> However for the implementation it is somewhat necessary. I'll run it locally.

It is not _absolutely_ necessary, when working on your application to act on webhooks received from Hyperswitch, you _can_ use something like the `smee-client` to forward webhooks to your development machine, if you don't prefer running Hyperswitch locally. You also have the option of running Hyperswitch locally, so yeah.
